src/modules/commander/failure_detector/FailureDetector.cpp

```diff
@@ -161,7 +176,43 @@ FailureDetector::updateExternalAtsStatus()
 		if (_ext_ats_failure_hysteresis.get_state()) {
 			_status |= FAILURE_EXT;
 		}
+
+		updated = true;
 	}
 
 	return updated;
-}
\ No newline at end of file
+}
+
+bool
+FailureDetector::updateEscsStatus(const vehicle_status_s &vehicle_status)
+{
+	hrt_abstime time_now = hrt_absolute_time();
+	bool updated(false);
+
+	if (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
+
+		esc_status_s esc_status{};
+		_sub_esc_status.copy(&esc_status);
+
+		int all_escs_armed = (1 << esc_status.esc_count) - 1;
+
+
+		_esc_failure_hysteresis.set_hysteresis_time_from(false, 300_ms);
+		_esc_failure_hysteresis.set_state_and_update(all_escs_armed != esc_status.esc_armed_flags, time_now);
+
+		if (_esc_failure_hysteresis.get_state() && !(_status & FAILURE_ARM_ESCS)) {
+			_status |= FAILURE_ARM_ESCS;
+			updated = true;
+		}
+
+	} else {
+		// reset ESC bitfield
+		_esc_failure_hysteresis.set_state_and_update(false, time_now);
+		_status &= ~FAILURE_ARM_ESCS;
+		updated = true;
```

Comment:
This should not always return true.

Reply:
fixed in here https://github.com/PX4/Firmware/pull/14766/files#diff-84f12e22ed352e6c0b6b55e22b6f0628R212